core/store/orm/orm.go

```diff
@@ -899,15 +899,45 @@ func (orm *ORM) TxAttempts(offset, limit int) ([]models.TxAttempt, int, error) {
 	return attempts, count, err
 }
 
-func (orm *ORM) UnconfirmedTxAttempts() ([]models.TxAttempt, error) {
+func (orm *ORM) HighestPricedUnconfirmedTxAttempts() ([]models.TxAttempt, error) {
 	var items []models.TxAttempt
+
+	subquery := orm.DB.
+		Select("tx_attempts.*").
+		Table("tx_attempts").
+		Joins("INNER JOIN txes ON txes.id = tx_attempts.tx_id").
+		Where("txes.confirmed = ?", false).
+		Order("tx_attempts.gas_price DESC").
+		SubQuery()
+
 	err := orm.DB.
-		Where("confirmed = false").
-		Find(&items).Error
+		Raw("SELECT * FROM ? GROUP BY tx_id", subquery).
+		Scan(&items).Error
+
+	if err != nil {
+		return nil, err
+	}
+
+	var ids []uint64
+	for i := range items {
+		ids = append(ids, items[i].TxID)
+	}
+
+	var txes []*models.Tx
+	err = orm.DB.
+		Where("id IN (?)", ids).
```

Comment:
I believe most DBs limit `IN` statements past a certain point. SQLite to 99 IIRC.